RingPublishingTracking/src/main/java/com/ringpublishing/tracking/internal/delegate/ReportDelegate.kt

```diff
@@ -1,50 +1,100 @@
-    }
-
-    fun reportPageView(partialReload: Boolean)
-    {
-        Logger.debug("App report PageView event partialReload=$partialReload")
-    }
-
-    fun reportContentPageView(contentMetadata: ContentMetadata, partialReload: ContentPageViewSource, delegate: Boolean, dataSource: RingPublishingTrackingKeepAliveDataSource)
-    {
-        Logger.debug("App report ContentPageView event contentMetadata=$contentMetadata partialReload=$partialReload delegate=$delegate contentKeepAliveDataSource$dataSource")
-    }
+	fun reportEvent(event: Event)
+	{
+		Logger.debug("App reported event $event")
+		val decoratedEvent = eventDecorator.decorate(event)
+		eventsService.addEvent(decoratedEvent)
+	}
+
+	fun reportEvents(events: List<Event>)
+	{
+		Logger.debug("App reported events $events")
+		eventsService.addEvents(events.map { eventDecorator.decorate(it) })
+	}
+
+	fun reportClick(elementName: String?)
+	{
+		Logger.debug("App report Click event elementName=$elementName")
+	}
+
+	fun reportUserAction(actionName: String, actionSubtypeName: String, parameters: Map<String, Any>)
+	{
+		Logger.debug("App report UserAction event actionName=$actionName actionSubtypeName=$actionSubtypeName parameters=$parameters")
+	}
+
+	fun reportUserAction(actionName: String, actionSubtypeName: String, parameters: String)
+	{
+		Logger.debug("App report UserAction event actionName=$actionName actionSubtypeName=$actionSubtypeName parameters=$parameters")
+	}
+
+	fun reportPageView(currentStructurePath: List<String>, partiallyReloaded: Boolean)
+	{
+		Logger.debug("App report PageView event currentStructurePath=$currentStructurePath partialReload=$partiallyReloaded")
+
+		configurationDelegate.updateCurrentStructurePath(currentStructurePath)
+		configurationDelegate.currentIsPartialView = partiallyReloaded
+
+		if (partiallyReloaded)
+		{
+			configurationDelegate.newSecondaryId()
+		}
+
+		// todo: should be eventDefaultAnalyticsSystemName ?
+		val event = Event(Constants.eventDefaultAnalyticsSystemName, EventName.PAGE_VIEW.text)
+
+		// todo: append parameters to this event
+
+		reportEvent(event)
+	}
+
+	fun reportContentPageView(
+	    contentMetadata: ContentMetadata,
+	    contentPageViewSource: ContentPageViewSource,
+	    currentStructurePath: List<String>,
+	    partiallyReloaded: Boolean,
+	    contentKeepAliveDataSource: RingPublishingTrackingKeepAliveDataSource,
+	)
+	{
+		Logger.debug("App report ContentPageView event " +
+				"contentMetadata=$contentMetadata " +
+				"contentPageViewSource=$contentPageViewSource " +
+				"currentStructurePath=$currentStructurePath" +
+				"partialReload=$partiallyReloaded ")
+
+		configurationDelegate.updateCurrentPublicationUrl(contentMetadata.publicationUrl)
+		configurationDelegate.updateCurrentStructurePath(currentStructurePath)
+		configurationDelegate.currentIsPartialView = partiallyReloaded
+
+		configurationDelegate.newPrimaryId()
```

Comment:
to tylko generujemy na nowo jak nie partial = false

Reply:
ok zmieniłem
a w reportPageView ma się generować na nowo tak:
if (partiallyReloaded)
		{
			configurationDelegate.newSecondaryId()
		}